src/main/java/com/ardaslegends/service/war/BattleService.java

```diff
@@ -36,82 +33,164 @@ public class BattleService extends AbstractService<Battle, BattleRepository> {
+                    throw BattleServiceException.defendingArmyIsMovingAway(defendingArmy);
+                }
+                log.debug("Defending army is moving but is still in the region for the next 24h");
+
+            }
+
+            defendingArmies.add(defendingArmy);
+            log.debug("Setting defending Faction to: [{}]", defendingArmy.getFaction().getName());
+            defendingFaction = defendingArmy.getFaction();
+            battleRegion = defendingArmy.getCurrentRegion();
         }
         else {
-            log.debug("Field battle boolean is set to false - fetching all stationed armies at claim build: [{}]", createBattleDto.ClaimBuildName());
-            log.trace("Calling getClaimBuildByName with name: [{}]", createBattleDto.ClaimBuildName());
-            ClaimBuild fetchedClaimBuild = claimBuildService.getClaimBuildByName(createBattleDto.ClaimBuildName());
-            defendingFaction = fetchedClaimBuild.getOwnedBy();
-            List<Army> stationedArmies = fetchedClaimBuild.getStationedArmies();
+            log.debug("Declared battle is claimbuild battle");
+            log.debug("Fetching all stationed armies at claimbuild: [{}]", createBattleDto.claimBuildName());
+            log.trace("Calling getClaimBuildByName with name: [{}]", createBattleDto.claimBuildName());
+            attackedClaimbuild = claimBuildService.getClaimBuildByName(createBattleDto.claimBuildName());
+            defendingFaction = attackedClaimbuild.getOwnedBy();
+            battleRegion = attackedClaimbuild.getRegion();
+            List<Army> stationedArmies = attackedClaimbuild.getStationedArmies();
+
+            log.debug("Checking if claimbuild [{}] is starter hamlet of faction [{}]", attackedClaimbuild.getName(), defendingFaction.getName());
+            if(attackedClaimbuild.getName().toLowerCase().endsWith("starter hamlet")) {
+                log.warn("Cannot declare battle on clamibuild [{}] because it is the starter hamlet of faction [{}]", attackedClaimbuild.getName(), defendingFaction.getName());
+                throw BattleServiceException.cannotAttackStarterHamlet();
+            }
+
+            log.debug("Calling pathfinder to find shortest way from regions [{}] to [{}]", attackingArmy.getCurrentRegion(), battleRegion);
+            path = pathfinder.findShortestWay(attackingArmy.getCurrentRegion(), attackedClaimbuild.getRegion(),executorPlayer,true);
+            log.debug("Path: [{}], duration: [{} days]", ServiceUtils.buildPathString(path), ServiceUtils.getTotalPathCost(path));
+
+            log.debug("Checking if claimbuild is reachable in 24 hours");
+            if(ServiceUtils.getTotalPathCost(path) > 24){
+                log.warn("Cannot declare battle - Claimbuild [{}] is not in 24 hour reach of attacking army [{}]", attackedClaimbuild.getName(), attackingArmy.getName());
+                throw BattleServiceException.battleNotAbleDueHours();
+            }
```

Comment:
Do we maybe want to extract these two conditions into separate methods. Because this method is gigantic.

Reply:
Thought about it too, but it somehow felt weird because there is no reusability aspect, we'd just split it up for readability. But I can do it if you want